packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -609,6 +622,7 @@ export class CardStack extends React.Component<Props, State> {
                   })
                 : STATE_TRANSITIONING_OR_BELOW_TOP;
             }
+            this.lastActivityStateForIndex[index] = isScreenActive;
```

Comment:
This makes render impure. We need to figure out an alternative way that keeps render pure.

Reply:
Do you have an idea what could we do to achieve it?

Reply:
Maybe let's have a call to understand the logic and come up with an idea

Reply:
I don't remember much more than what is written in the description of the PR, but we can do that for sure 😅